src/Npgsql/NpgsqlBinaryImporter.cs

```diff
@@ -301,9 +302,22 @@ public void Complete()
                 _buf.EndCopyMode();
 
                 _connector.SendMessage(CopyDoneMessage.Instance);
-                Expect<CommandCompleteMessage>(_connector.ReadMessage());
+
+                var msg1 = _connector.ReadMessage();
+                Expect<CommandCompleteMessage>(msg1);
                 Expect<ReadyForQueryMessage>(_connector.ReadMessage());
                 _state = ImporterState.Committed;
+
+                CommandCompleteMessage cmdComplete;
+                switch (msg1.Code)
```

Comment:
Why check the code here? You've made sure with `Expect<>` above that it's CommandComplete, no?

Reply:
These are all very helpful and professional feedback comments!
Thanks very much for taking the time to provide the quality check, given your deep knowledge and stewardship of this fine project.
 
I think the simple answer is: a) I can't quite remember since I wrote this in a hurry to solve a problem in May/June. b) I think for expediency and completeness, I was re-using a similar "property checking" structure either from NpgsqlBinaryExporter or another class, that clearly required a lot more checks for the simpler case in NpgsqlBinaryImpoter.

(Continued in next code review comment)